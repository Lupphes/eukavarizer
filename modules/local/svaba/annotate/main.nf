process SVABA_ANNOTATE {
    tag "$meta.id"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container 'quay.io/biocontainers/python:3.10'

    input:
    tuple val(meta), path(vcf)

    output:
    tuple val(meta), path("*.reclassified.vcf"), emit: vcf
    path "versions.yml"                        , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    gunzip -c ${vcf} > input.vcf
    svaba_annotate.py ${args} input.vcf > ${prefix}.reclassified.vcf

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$( python --version 2>&1 | sed 's/Python //g' )
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.reclassified.vcf

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$( python --version 2>&1 | sed 's/Python //g' )
    END_VERSIONS
    """
}
