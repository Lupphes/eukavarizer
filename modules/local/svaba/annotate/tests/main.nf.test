nextflow_process {

    name "Test Process SVABA_ANNOTATE"
    script "../main.nf"
    process "SVABA_ANNOTATE"
    tag "svaba"
    tag "svaba/annotate"
    tag "modules"
    tag "modules_local"

    test("Annotate SVABA VCF - stub") {

        options '-stub'

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = channel.of([
                    [ id: 'test' ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/vcf/test_annotate.vcf.gz')
                ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() },
                { assert process.out.vcf },
                { assert process.out.vcf.get(0).get(1).toString().endsWith('.vcf') },
                { assert process.out.versions }
            )
        }
    }

    test("Annotate with different sample ID - stub") {

        options '-stub'

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = channel.of([
                    [ id: 'sample_123' ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/vcf/test_annotate.vcf.gz')
                ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() },
                { assert process.out.vcf.get(0).get(1).toString().contains('sample_123') },
                { assert process.out.versions }
            )
        }
    }
}
