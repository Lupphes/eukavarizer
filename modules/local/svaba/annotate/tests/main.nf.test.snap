{
    "Annotate SVABA VCF - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.reclassified.vcf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,4c8c79218294527a4dff92866ce5de78"
                ],
                "vcf": [
                    [
                        {
                            "id": "test"
                        },
                        "test.reclassified.vcf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,4c8c79218294527a4dff92866ce5de78"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.04.8"
        },
        "timestamp": "2025-10-20T22:36:48.290011066"
    },
    "Annotate with different sample ID - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "sample_123"
                        },
                        "sample_123.reclassified.vcf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,4c8c79218294527a4dff92866ce5de78"
                ],
                "vcf": [
                    [
                        {
                            "id": "sample_123"
                        },
                        "sample_123.reclassified.vcf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,4c8c79218294527a4dff92866ce5de78"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.04.8"
        },
        "timestamp": "2025-10-20T22:36:56.592645426"
    }
}